src/controllers/update/overseas.entity.query.controller.ts

```diff
@@ -69,3 +59,26 @@ function createEntityNumberError(entityNumber: string): any {
   return errors;
 }
 
+const renderGetPageWithError = (res: Response, entityNumber: any) => {
+  const errors = createEntityNumberError(entityNumber);
+  return res.render(config.OVERSEAS_ENTITY_QUERY_PAGE, {
+    backLinkUrl: config.UPDATE_LANDING_PAGE_URL,
+    templateName: config.OVERSEAS_ENTITY_QUERY_PAGE,
+    [EntityNumberKey]: entityNumber,
+    errors
+  });
+};
+
+const addOeToApplicationData = async (req: Request, appData: ApplicationData, entityNumber: any, companyProfile: CompanyProfile) => {
+  resetEntityUpdate(appData);
+  appData.entity_name = companyProfile.companyName;
+  appData.entity_number = entityNumber;
+  appData.entity = mapCompanyProfileToOverseasEntity(companyProfile);
+  if (appData.update) {
+    appData.update.date_of_creation = mapInputDate(companyProfile.dateOfCreation);
+  }
+
+  await retrieveBoAndMoData(req, appData);
```

Comment:
I think we have been careful that the only real change is to call the await retrieveBoAndMoData here.

Reply:
That is essentially the only change, I've just re-factored/moved the existing code from the POST controller into it's own method as the POST controller was growing quite messy and large.